File: frontend/src/types/abTestResult.ts
export interface ABTestResult {
  id: number
  testId: number
  testName: string
  status: 'RUNNING' | 'COMPLETED' | 'PAUSED' | 'CANCELLED'
  startDate: string
  endDate?: string
  variants: TestVariant[]
  winner?: string
  confidence: number
  metric: string
  platform: string
  createdAt: string
}

export interface TestVariant {
  id: string
  name: string
  description: string
  thumbnailUrl?: string
  views: number
  clicks: number
  ctr: number
  avgWatchTime: number
  engagement: number
  conversions: number
  isControl: boolean
  isWinner: boolean
}

export interface TestTimeline {
  date: string
  variantA: number
  variantB: number
  variantC?: number
}

export interface ABTestResultSummary {
  totalTests: number
  runningTests: number
  completedTests: number
  avgConfidence: number
  avgImprovement: number
}

File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/abtestresult/ABTestResult.kt
package com.ongo.domain.abtestresult

import java.math.BigDecimal
import java.time.LocalDateTime

data class ABTestResult(
    val id: Long = 0,
    val workspaceId: Long,
    val testId: Long? = null,
    val testName: String,
    val status: String = "RUNNING",
    val startDate: LocalDateTime,
    val endDate: LocalDateTime? = null,
    val winner: String? = null,
    val confidence: BigDecimal = BigDecimal.ZERO,
    val metric: String = "CTR",
    val platform: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val updatedAt: LocalDateTime = LocalDateTime.now(),
)
